src/api/exchange/ExchangeApi.ts

```diff
@@ -109,6 +121,38 @@ export class ExchangeApiImpl extends DepositApiImpl implements ExchangeApi {
     return decodeAuctionElements(encodedOrders)
   }
 
+  public async getOrdersPaginated({
+    userAddress,
+    networkId,
+    offset,
+    pageSize,
+  }: GetOrdersPaginatedParams): Promise<GetOrdersPaginatedResult> {
+    const contract = await this._getContract(networkId)
+    const _pageSize = pageSize || DEFAULT_ORDERS_PAGE_SIZE
+
+    log(
+      `[ExchangeApiImpl] Getting Orders Paginated for account ${userAddress} with offset ${offset} and pageSize ${_pageSize}`,
+    )
+
+    const encodedOrders = await contract.methods.getEncodedUserOrdersPaginated(userAddress, offset, _pageSize).call()
+
+    // is null if Contract returns empty bytes
+    if (!encodedOrders) return { orders: [] }
+
+    const orders = decodeAuctionElements(encodedOrders, offset)
+
+    let nextIndex: number | undefined
+    if (orders.length < _pageSize) {
+      // no more pages left, indicate by not returning `nextIndex`
+      nextIndex = undefined
+    } else {
+      // this page was full, keep going
+      nextIndex = Number(orders[orders.length - 1].id) + 1
```

Comment:
I like this part where we return the nextIndex, although I would follow a slightly different pattern.

Instead of querying `pageSize results` (the ones you need),  query `pageSize + 1` then. 
For the result, you'll discard the element `pageSize + 1` and return it's id as `nextIndex`. 

This way, you know for sure if you have more pages or not. With the current approach, if for example you query 10 results and there's 10 items, you'll return `nextIndex=11` instead of `nextIndex =undefined`

Reply:
I like that, will do it.